apps/aecore/src/aec_target.erl

```diff
@@ -35,10 +35,15 @@ recalculate(Header, InitialHeader) ->
     CurrentRate = mining_rate_between(InitialHeader, Header),
     aec_pow:recalculate_target(HeaderTarget, ExpectedRate, CurrentRate).
 
--spec verify(header(), header()) -> ok | {error, target_too_high}.
-verify(Header, InitialHeader) ->
+-spec verify(header(), header(), header()) -> ok | {error, target_too_high}.
+verify(Header, PrevHeader, InitialHeader) ->
     HeaderTarget = aec_headers:target(Header),
-    ExpectedTarget = recalculate(Header, InitialHeader),
+    PrevTarget   = aec_headers:target(PrevHeader),
+
+    %% Block candidate target was computed using the target of the previous block
+    %% we have to do the same thing here.
+    ExpectedTarget = recalculate(Header#header{ target = PrevTarget }, InitialHeader),
```

Comment:
Ì am slightly uneasy about the use of the header record directly here.
Perhaps create a set_target function in aec_header.erl